Event.php

```diff
@@ -186,10 +190,23 @@ public static function unregister($eventId, $hard = false)
      * @param   mixed   $callable    Callable.
      * @return  \Hoa\Event\Event
      */
-    public function attach($callable)
+    public function attach($callable, $priority = 0)
     {
-        $callable                              = xcallable($callable);
-        $this->_callable[$callable->getHash()] = $callable;
+        $callable                 = xcallable($callable);
+        $hash                     = $callable->getHash();
+        $this->_callable[$hash]   = $callable;
```

Comment:
I guess we can change the `_callable` data structure for a heap or a priority queue. You might take a look at [`SplHeap`](https://secure.php.net/manual/en/class.splheap.php), [`SplMaxHeap`](https://secure.php.net/manual/en/class.splmaxheap.php), [`SplMinHeap`](https://secure.php.net/manual/en/class.splminheap.php) or [`SplPriorityQueue`](https://secure.php.net/manual/en/class.splpriorityqueue.php).